Tests/TestUtils/XCTestCase+AnyCodableAsserts.swift

```diff
@@ -398,186 +469,127 @@ extension XCTestCase {
 
                     Key path: \#(keyPathAsString(keyPath))
                 """#, file: file, line: line)
+                }
+                return false
             }
-            return false
-        }
+            var actualIndexes = Set(0..<actual.count)
+            var expectedIndexes = Set(0..<expected.count)
+            var wildcardIndexes: Set<Int>
+
+            // Collect all the keys from `pathTree` that either:
+            // 1. Mark the path end (where the value is a `String`), or
+            // 2. Contain the asterisk (*) character.
+            let pathEndKeys = pathTree?.filter { key, value in
+                value is String || key.contains("*")
```

Comment:
This might not work as expected if the key has * in it eg: `awesome*key`. Should we check startsWith and EndsWith '*'? applies to Android code as well.

Reply:
Yes, I saw that we will drop it while after casting, but I was thinking if we could set a preferred way to do this. Like it has to start with * or end with *. So we don't do not necessary casting and also keep it standardized.

How about `"1*2"` for the above test?